### ISSUE TITLE
SMS_RECEIVED priority is set too high

### ISSUE THREAD

**Issue by Slimithy** [signalapp/Signal-Android#6933]
I have:
- [x] searched open and closed issues for duplicates
----------------------------------------

### SMS_RECEIVED priority is set at 1001, shouldn't exceed 999
- Is causing problem with another app
- Keeping SMS messages from being received other app, which is default SMS app

**Device:** LGMS330
**Android version:** 5.1.1
**Signal version:** 4.10.3

**Comment by johanw666** [comment]
Which other app? Which Android version? The sms handling has been changed in Android 4.4: since then, ALL apps that have the read sms permission can see it and can decide what to do with it.

I think the maker of this other app might be interested in reading https://stackoverflow.com/questions/35970142/broadcastreceiver-sms-received-not-working-on-new-devices , and since Silence also had this issue reading this thread https://github.com/SilenceIM/Silence/issues/6 might also help.

In Silence, we boosted the priority to 1002 to prevent old Signal versions on Android 4.3 and below to complain about encrypted sms messages being deprecated. After Signal removed the code to do this completely the issue became moot anyway.

**Comment by Slimithy** [comment]
The other app it is hide my text and invisible second text application made by second phone apps but with signal having a 1001 it will prevent the other app from even receiving 9 cases but in some

On Sep 7, 2017 8:52 AM, "johanw666" <notifications@github.com<mailto:notifications@github.com>> wrote:

Which other app? Which Android version? The sms handling has been changed in Android 4.4: since then, ALL apps that have the read sms permission can see it and can decide what to do with it.

I think the maker of this other app might be interested in reading https://stackoverflow.com/questions/35970142/broadcastreceiver-sms-received-not-working-on-new-devices , and since Silence also had this issue reading this thread SilenceIM/Silence#6<https://github.com/SilenceIM/Silence/issues/6> might also help.

In Silence, we boosted the priority to 1002 to prevent old Signal versions on Android 4.3 and below to complain about encrypted sms messages being deprecated. After Signal removed the code to do this completely the issue became moot anyway.
